src/TokenHandler/AccountManager.mo

```diff
@@ -23,7 +23,10 @@ module {
     #feeUpdated : { old : Nat; new : Nat };
     #newDeposit : Nat;
     #consolidated : { deducted : Nat; credited : Nat };
-    #consolidationError : ICRC1.TransferError or { #CallIcrc1LedgerError };
+    #consolidationError : ICRC1.TransferError or {
+      #CallIcrc1LedgerError;
+      #ExtendedBadFee : { original_fee : Nat; expected_fee : Nat };
+    };
```

Comment:
This makes the code too complicated. Please don't report the original fee from the inner function `processConsolidationTransfer` to the outer function `consolidate`. Keep their interaction as before. The outer function can save original fee itself for use in the continuation. Just put a line 
```
let originalFee : Nat = fee_;
```
in `consolidate` before it calls `processConsolidationTransfer`.